iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java

```diff
@@ -33,29 +33,29 @@ public LoadDetailXmlSerializer() {
 
     @Override
     public void write(LoadDetail detail, XmlWriterContext context) throws XMLStreamException {
-        XmlUtil.writeFloat("fixedActivePower", detail.getFixedActivePower(), context.getWriter());
-        XmlUtil.writeFloat("fixedReactivePower", detail.getFixedReactivePower(), context.getWriter());
-        XmlUtil.writeFloat("variableActivePower", detail.getVariableActivePower(), context.getWriter());
-        XmlUtil.writeFloat("variableReactivePower", detail.getVariableReactivePower(), context.getWriter());
+        XmlUtil.writeDouble("fixedActivePower", detail.getFixedActivePower(), context.getWriter());
+        XmlUtil.writeDouble("fixedReactivePower", detail.getFixedReactivePower(), context.getWriter());
+        XmlUtil.writeDouble("variableActivePower", detail.getVariableActivePower(), context.getWriter());
+        XmlUtil.writeDouble("variableReactivePower", detail.getVariableReactivePower(), context.getWriter());
     }
 
     @Override
     public LoadDetail read(Load load, XmlReaderContext context) {
-        float fixedActivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "fixedActivePower");
-        if (Float.isNaN(fixedActivePower)) {
-            fixedActivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad1ActivePower");
+        double fixedActivePower = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "fixedActivePower");
+        if (Double.isNaN(fixedActivePower)) {
+            fixedActivePower = XmlUtil.readDoubleAttribute(context.getReader(), "subLoad1ActivePower");
         }
-        float fixedReactivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "fixedReactivePower");
-        if (Float.isNaN(fixedReactivePower)) {
-            fixedReactivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad1ReactivePower");
+        double fixedReactivePower = XmlUtil.readDoubleAttribute(context.getReader(), "fixedReactivePower");
```

Comment:
This should be `readOptionalDoubleAttribute` here

Reply:
Used readOptionalDoubleAttribute before NaN checking